source_lists/libwesnoth_widgets

```diff
@@ -13,6 +13,7 @@ gui/widgets/minimap.cpp
 gui/widgets/multi_page.cpp
 gui/widgets/multiline_text.cpp
 gui/widgets/multimenu_button.cpp
+gui/widgets/options_button.cpp
```

Comment:
Use `./add_source_file` script to add the new source file, instead of adding it manually. This ensure it gets correctly add in XCode and CodeBlocks projectfiles.
(Note: add_source_file adds widgets to `libwesnoth`, which you'll need to manually remove. The placement here is correct.)

Reply:
I did add it with ./add_source_file, though it is possible I added it manually before I saw your comment (I did not know about add_source_file).

I don't see any widgets in libwesnoth.

I notice the Mac builds are failing with a linker error on my new widget.  Do I need to push anything from projectfiles/ ?  I don't have any idea what XCode and CodeBlocks are, but their files did change when I ran add_source_file.

Reply:
> Do I need to push anything from projectfiles/ ? I don't have any idea what XCode and CodeBlocks are, but their files did change when I ran add_source_file.

Yes, you need to push anything under `projectfiles/` that got changed. The CI failure is related to not having XCode projectfile updated, which is needed for MacOS builds.

Reply:
> I don't see any widgets in libwesnoth.

That's good, but do a check to see if anything under `source_lists` got changed. A source file should only be listed in one source list file (in this case `libwesnoth-widgets`).
Also, you can run the whitespace fixer with `./utils/CI/fix_whitespace.sh` and push any changed files for the whitespace CI check to succeed.

Reply:
Great, thanks.  I must have manually edited that file and then run add_source_file.  I think I have that cleaned up now.

Can you take a look at the current build errors?  I'm lost there.

1) Mac:
```
In file included from /Users/runner/work/wesnoth/wesnoth/src/gui/widgets/options_button.cpp:18:
/Users/runner/work/wesnoth/wesnoth/src/gui/widgets/options_button.hpp:162:7: error: private field 'persistent_' is not used [-Werror,-Wunused-private-field]
        bool persistent_;

```
I sure looks to me like it's used.

2) Ubuntu MP tests:
I can't tell what's failing.  I can't view the log file on github (or download), I can just see the first 1.5million or so repeated worthless messages (which pushes firefox up around 5.5 of my 8GB).

I tried running the test on my laptop, but I couldn't see anything except it wanted to open a gui thing that was 633 wide with 600 available, but no clue as to what.  I tried adding some gui logging, but running that script on my laptop I never get anything like the same results twice, so I have no idea what it's choking on.

Reply:
> I sure looks to me like it's used.

You have a variable named `persistent_` in both the main class and the builder class. The builder class' version is not used.

Reply:
Ah, I could not see that no matter how many times I looked. Now that you point it out it's obvious, thank you.

I found a little more on the MP test, the error I mentioned was something that is completely local to my environment (I'm playing with a gui2 sidebar and it was too tall for the MP test which is running in 800x600).  So that shouldn't be happening on github.

what I do see now locally is the test runs, but I don't get the millions of turns, I get a `error server: <unknown address>	Connection reset by peer`.  As this looks like it happens exactly 240 seconds from the time the second player joins, I'm thinking tcp_time_wait (or whatever it is on linux, I know on Solaris at least 240 was the default and I think that's kind of "recommended" -- or was, it's been a long time).

I know it probably doesn't matter how the test runs on MY laptop, but since I can't see where it's failing on github it's all I really have to work with.
[wesnothd.log](https://github.com/user-attachments/files/16598700/wesnothd.log)